projects/observability/src/pages/explorer/explorer.component.ts

```diff
@@ -167,10 +167,7 @@ export class ExplorerComponent {
     this.resultsExpanded$ = this.preferenceService.get(ExplorerComponent.RESULTS_EXPANDED_PREFERENCE, true);
     this.resultsDashboard$ = this.explorerDashboardBuilder.resultsDashboard$;
     this.vizDashboard$ = this.explorerDashboardBuilder.visualizationDashboard$;
-    this.initialState$ = activatedRoute.queryParamMap.pipe(
-      take(1),
-      map(paramMap => this.mapToInitialState(paramMap))
-    );
+    this.initialState$ = activatedRoute.queryParamMap.pipe(map(paramMap => this.mapToInitialState(paramMap)));
```

Comment:
have you verified this doesn't lead to any cycles that would need to be handled (applying initial state updates the url, which applies state again)?

Reply:
Checked using `tap` on the observable. There are no cycles as such, only thing is that as `initialState$` is subscribed at the outermost div on the template, it re-renders all the components everytime we apply a new query instead of updating just the states.